File: src/easy/array-pair-sum.ts
export const arrayPairSum = (nums: number[]): number => {
	let total = 0;
	nums.sort((a, b) => b - a);
	for (let i = 0; i < nums.length - 1; i += 2) {
		total += Math.min(nums[i], nums[i + 1]);
	}
	return total;
};

File: src/easy/array_pair_sum.zig
const std = @import("std");

const Allocator = std.mem.Allocator;

pub fn arrayPairSum(comptime T: type, allocator: Allocator, nums: []const T) ?T {
    var total: T = 0;
    const arr = allocator.dupe(T, nums) catch return null;
    defer allocator.free(arr);
    std.mem.sort(T, arr, {}, std.sort.asc(T));
    std.mem.reverse(T, arr);
    var index: usize = 0;
    while (index < nums.len - 1) : (index += 2) {
        total += @min(arr[index], arr[index + 1]);
    }
    return total;
}
